cli/azd/pkg/pipeline/pipeline_manager.go

```diff
@@ -781,3 +756,183 @@ func (pm *PipelineManager) savePipelineProviderToEnv(
 	}
 	return nil
 }
+
+func (pm *PipelineManager) checkAndPromptForProviderFiles(
+	ctx context.Context, repoRoot, pipelineProvider string, infraProvider string) error {
+	if pipelineProvider == "" {
+		return nil // No provider, no need to check for files
+	}
+
+	log.Printf("Checking for provider files for: %s", pipelineProvider)
+
+	hasGitHubYml := osutil.FileExists(filepath.Join(repoRoot, gitHubYml))
+	hasAzDevOpsYml := osutil.FileExists(filepath.Join(repoRoot, azdoYml))
+
+	log.Printf("hasGitHubYml: %t, hasAzDevOpsYml: %t", hasGitHubYml, hasAzDevOpsYml)
+
+	shouldPrompt := false
+
+	switch pipelineProvider {
+	case gitHubLabel:
+		if !hasGitHubYml {
+			log.Printf("GitHub YAML not found, prompting for creation")
+			shouldPrompt = true
+		}
+	case azdoLabel:
+		if !hasAzDevOpsYml {
+			log.Printf("Azure DevOps YAML not found, prompting for creation")
+			shouldPrompt = true
+		}
+	default:
+		return fmt.Errorf("%s is not a known pipeline provider", pipelineProvider)
+	}
+
+	if shouldPrompt {
+		err := pm.promptForCiFiles(ctx, pipelineProvider, infraProvider, repoRoot)
+		if err != nil {
+			return err
+		}
+	}
+
+	// Check for files in the selected provider directory before exiting the method
+	providerDir := ""
+	providerDirDisplay := ""
+	switch pipelineProvider {
+	case gitHubLabel:
+		providerDir = filepath.Join(repoRoot, gitHubWorkflowsDirectory)
+		providerDirDisplay = gitHubWorkflowsDirectory
+	case azdoLabel:
+		providerDir = filepath.Join(repoRoot, azdoPipelinesDirectory)
+		providerDirDisplay = azdoPipelinesDirectory
+	}
+
+	log.Printf("Checking if directory %s is empty", providerDir)
+	isEmpty, err := osutil.IsDirEmpty(providerDir, true)
```

Comment:
@vhvb1989 - Is it okay to have an empty .github/workflows folder?  I know you said we need a file for .azdo, wondering if that is true for GH as well.